src/fparser/two/C99Preprocessor.py

```diff
@@ -0,0 +1,561 @@
+#!/usr/bin/env python
+
+# All rights reserved.
+
+# Modifications made as part of the fparser project are distributed
+# under the following license:
+
+# Redistribution and use in source and binary forms, with or without
+# modification, are permitted provided that the following conditions are
+# met:
+
+# 1. Redistributions of source code must retain the above copyright
+# notice, this list of conditions and the following disclaimer.
+
+# 2. Redistributions in binary form must reproduce the above copyright
+# notice, this list of conditions and the following disclaimer in the
+# documentation and/or other materials provided with the distribution.
+
+# 3. Neither the name of the copyright holder nor the names of its
+# contributors may be used to endorse or promote products derived from
+# this software without specific prior written permission.
+
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+# "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+# LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR
+# A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT
+# HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+"""C99 Preprocessor Syntax Rules.
+"""
+# Author: Balthasar Reuter <balthasar.reuter@ecmwf.int>
+# Based on previous work by Martin Schlipf (https://github.com/martin-schlipf)
+# First version created: Jan 2020
+
+import re
+
+from fparser.two import pattern_tools as pattern
+from fparser.two.utils import (Base, StringBase, InternalError)
```

Comment:
It is actually superfluous in this case - I guess I had more imports in that statement in an earlier stage, which required a line break, and never removed the parenthesis when no longer needed.

It is accepted standard (introduced in [PEP328](https://www.python.org/dev/peps/pep-0328/)) and was partially born out of PEP8's [preference of parentheses instead of backslash](https://www.python.org/dev/peps/pep-0008/#maximum-line-length) for line continuation:
> The preferred way of wrapping long lines is by using Python's implied line continuation inside parentheses, brackets and braces. Long lines can be broken over multiple lines by wrapping expressions in parentheses. These should be used in preference to using a backslash for line continuation.

Personally, I don't always see it as advantageous as it mandates indentation on following lines to align with the opening parenthesis, which limits the useful line length significantly when the import path is long.